tendermint/src/lite/verifier.rs

```diff
@@ -34,8 +44,38 @@ where
     Ok(())
 }
 
-// Validate the validators and commit against the header.
-fn validate_vals_and_commit<H, V, C>(header: H, commit: &C, vals: &V) -> Result<(), Error>
+/// Captures the skipping condition, i.e., it defines when we can trust the header
+/// h2 based on header h1.
+/// Note that h1 and h2 have already passed basic validation by calling  `verify`.
+/// `Error::InsufficientVotingPower`is returned when there is not enough intersection
+/// between validator sets to have skipping condition true.
+pub fn check_support<S, V, L>(
+    h1: &S,
+    h1_next_vals: &V,
+    h2: &S,
+    trust_threshold: L,
+    trusting_period: Duration,
+    now: SystemTime,
+) -> Result<(), Error>
+where
+    S: SignedHeader,
+    V: ValidatorSet,
+    L: TrustThreshold,
+{
+    if let Err(err) = expired(h1.header(), trusting_period, now) {
```

Comment:
Is there a way to get rid of the `if let` here and just oneline it - ie. return if its an error, otherwise do nothing? `?` ?

Reply:
Yes, but we'd need to provide a way to from `SystemTimeError` to lite::Error. Tracked in #96.